### PR TITLE
batman-adv-legacy: remove batctl v2013.4

### PR DESCRIPTION
**Pull request by T-X**
With the netlink functions backported batman-adv-legacy works fine with
the upstream batctl. Therefore there is no more need to build and
install the old batctl v2013.4 version.

### CODE REVIEW

File: net/batman-adv-legacy/Makefile [hunk 1]
```diff
@@ -112,8 +66,6 @@ endef
 
 define Build/Patch
```
Comment (NeoRaider) [inline net/batman-adv-legacy/Makefile]: We should be able to remove all these explicit Build/* steps as well and use the defaults. Also drop the batctl patches in the *patches* subdir.

**Comment by T-X** [comment]
Changelog v2:
* removed batctl patches
* remove explicit Build/{DoPatch,Patch,Prepare} instructions (still needed this Build/Compile variant, otherwise the build would fail)

**Comment by NeoRaider** [comment]
>     * remove explicit Build/{DoPatch,Patch,Prepare} instructions (still needed this Build/Compile variant, otherwise the build would fail)
Just rename MAKE_BATMAN_ADV_ARGS to MAKE_FLAGS, then you can use the default Build/Compile defined in `include/package-defaults.mk`. You can also drop ARCH, which is already in MAKE_FLAGS by default.

**Comment by T-X** [comment]
Changelog v3:
* renamed MAKE_BATMAN_ADV_ARGS to MAKE_FLAGS
* removed Build/Compile

**Comment by NeoRaider** [comment]
> You can also drop ARCH, which is already in MAKE_FLAGS by default.

Oops, I just noticed that ARCH is defined in the defaults, but not with the correct value. Please add it back, then we can get this merged.

**Comment by T-X** [comment]
Changelog v4:
* readded "$ARCH" in MAKE_FLAGS

**Pull request merged**